src/data/mv_polynomial/basic.lean

```diff
@@ -1082,6 +1082,9 @@ lemma aeval_eq_eval₂_hom (p : mv_polynomial σ R) :
 
 @[simp] lemma aeval_C (r : R) : aeval f (C r) = algebra_map R S₁ r := eval₂_C _ _ _
 
+-- `simp` doesn't much like `map_nat_cast`.
```

Comment:
What does this comment mean? We certainly want `simp` to trigger `map_nat_cast` simplifications. If that is not possible for certain reasons (out_params, or something like that??) then I fear this is a regression.

Reply:
Most of the time, it works (even within simp), but I had to fix a couple that didn't; this version is harder to elaborate, so in very meta-variably states (like term proofs) this new map_nat_cast can just fail to trigger. However, as a stop-gap we can just put the old versions back as a corollary of this lemma if it starts being an issue. This comment was made after much frustration...

Reply:
This is now no longer true after the kernel improvements.